src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,166 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> element = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = element;
+        } else {
+            tail.next = element;
+            element.prev = tail;
+            tail = element;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAddition(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> current = (index < size / 2) ? getNodeFromHead(index) : getNodeFromTail(index);
+        Node<T> newElement = new Node<>(current.prev, value, current);
+        if (current.prev != null) {
+            current.prev.next = newElement;
+        } else {
+            head = newElement;
+        }
+        current.prev = newElement;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = (index < size / 2) ? getNodeFromHead(index) : getNodeFromTail(index);
+        return current.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = (index < size / 2) ? getNodeFromHead(index) : getNodeFromTail(index);
+        T oldValue = current.value;
+        current.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = (index < index / 2) ? getNodeFromHead(index) : getNodeFromTail(index);
```

Comment:
There is a bug in the condition for the ternary operator. It should check if the index is less than `size / 2`, not `index / 2`.